### PR TITLE
Add a native destop notification when a new question enters the queue.

### PR DESCRIPTION
**Pull request by ngwattcos**
### Summary 
Adds a native desktop notification when a new question enters the queue. Use 'react-push-notification' to create a notification when.

Also solves issue #340 where a notification is pushed whenever the SessionView component re-renders, whether 1) a question is added (good), 2) web page is resized (bad), or session question state changes/is selected (bad). The problem with checking ```cachedPrevQuestions === questions``` is that, on re-renders, 'questions' is re-populated by objects from Firestore with the exact same data. So 'questions' is a different reference.

This solved by comparing the current list of question ids with the cached question ids in ```prevQuestSet``` - we only care if current list of `questions` has any new questionIds that the ```prevQuestSet``` does not have.

#### The following scenarios were tested:
#####A notification should show up when....
* every time a single question is entered into the queue. Only one notification on the TA/prof's view should pop up.
* a new question is entered into the queue, but questions already exist in the queue

#####A notification should not show up when...
* a student creates a question, but removes their question
* From TA/prof user: resize the window to half width. No notifications should pop up.
* From TA/prof user: click 'Assign to me'. No notifications should pop up.

#####This does not crash...
Safari ios.

### Breaking Changes 
You just need to run ```yarn install``` to get react-push-notification.

- [ ] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

### CODE REVIEW

**Comment by SamChou19815** [comment]
@ngwattcos have you tested in on mobile iOS to ensure it didn't crash?

**Comment by ngwattcos** [comment]
@SamChou19815 I just tested in ios Safari and nothing breaks. Push notifications aren't supported in iOS Safari.

File: types/react-web-notification/index.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module 'react-web-notification';
```
Comment (SamChou19815) [inline types/react-web-notification/index.d.ts:1]: This package is written in TS. https://github.com/fabioshub/react-push-notification
Why are you declaring this?

Comment (ngwattcos) [reply types/react-web-notification/index.d.ts:1]: Yikes, I forgot to delete this. This was for an older package that didn't have an @types already. I stopped using that package and stashed my changes so I'm not sure why this ended up getting pushed...

**Pull request merged**